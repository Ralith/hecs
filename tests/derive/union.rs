use hecs::{Bundle, Query};

#[derive(Query)]
union Foo {
    u8: u8,
}

#[derive(Bundle)]
union Bar {
    u8: u8,
}

fn main() {}
