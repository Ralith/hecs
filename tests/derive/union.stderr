error: derive(Query) may only be applied to structs
 --> $DIR/union.rs:4:7
  |
4 | union Foo {
  |       ^^^

error: derive(Bundle) does not support enums or unions
 --> $DIR/union.rs:9:7
  |
9 | union Bar {
  |       ^^^
